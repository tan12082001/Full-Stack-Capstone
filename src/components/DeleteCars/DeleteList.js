import React from 'react';
import { useSelector } from 'react-redux';
import EachCarDelete from './EachCarDelete';

const DeleteList = () => {
  const cars = useSelector((state) => state.cars.cars);

  return (
    <>
      <div className="delete-cars-list">
        <h1 id="delete-list-id">Cars List</h1>
        <table>
          <thead>
            <tr>
              <th id="delete-list-car-title">Car Title</th>
              <th id="delete-list-car-status">Status</th>
            </tr>
          </thead>
          <tbody>
            {cars.map((car) => (
              <EachCarDelete key={car.id} car={car} />
            ))}
          </tbody>
        </table>
      </div>
    </>
  );
};

export default DeleteList;
