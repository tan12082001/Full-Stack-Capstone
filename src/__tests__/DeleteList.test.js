import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import DeleteList from '../components/DeleteCars/DeleteList';

const mockStore = configureMockStore();

describe('DeleteList Component', () => {
  it('renders cars present to delete', () => {
    const store = mockStore({
      authentication: {
        authenticatedUser: {
          username: 'testuser',
        },
        status: 'succeeded',
      },
      cars: {
        cars: [
          {
            id: 1,
            name: 'Test Vehicle',
          },
          {
            id: 2,
            name: 'Vehicle two',
          },
          {
            id: 3,
            name: 'Vehicle three',
          },
        ],
      },
    });

    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <DeleteList />
        </MemoryRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
