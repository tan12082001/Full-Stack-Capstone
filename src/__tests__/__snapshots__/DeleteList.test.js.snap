// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteList Component renders cars present to delete 1`] = `
<div>
  <div
    class="delete-cars-list"
  >
    <h1
      id="delete-list-id"
    >
      Cars List
    </h1>
    <table>
      <thead>
        <tr>
          <th
            id="delete-list-car-title"
          >
            Car Title
          </th>
          <th
            id="delete-list-car-status"
          >
            Status
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="delete-list-each-car"
        >
          <td>
            Test Vehicle
          </td>
          <td>
            <button
              class="remove-mark"
              type="button"
            >
              DELETE
            </button>
          </td>
        </tr>
        <tr
          class="delete-list-each-car"
        >
          <td>
            Vehicle two
          </td>
          <td>
            <button
              class="remove-mark"
              type="button"
            >
              DELETE
            </button>
          </td>
        </tr>
        <tr
          class="delete-list-each-car"
        >
          <td>
            Vehicle three
          </td>
          <td>
            <button
              class="remove-mark"
              type="button"
            >
              DELETE
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
